src/search_request.c

```diff
@@ -179,6 +206,47 @@ RSSearchRequest *ParseRequest(RedisSearchCtx *ctx, RedisModuleString **argv, int
   return NULL;
 }
 
+ReturnedField *FieldList_AddField(FieldList *fields, const char *name) {
+  size_t foundIndex = -1;
+  for (size_t ii = 0; ii < fields->numRawFields; ++ii) {
+    if (!strcasecmp(fields->rawFields[ii], name)) {
+      foundIndex = ii;
+      break;
+    }
+  }
+
+  if (foundIndex == -1) {
+    foundIndex = fields->numRawFields;
+    fields->rawFields =
+        realloc(fields->rawFields, sizeof(*fields->rawFields) * ++fields->numRawFields);
+    fields->rawFields[foundIndex] = strdup(name);
+  }
+
+  fields->fields = realloc(fields->fields, sizeof(*fields->fields) * ++fields->numFields);
+  ReturnedField *ret = fields->fields + (fields->numFields - 1);
+  memset(ret, 0, sizeof *ret);
+  ret->nameIndex = foundIndex;
+  return ret;
+}
+
+ReturnedField *FieldList_AddFieldR(FieldList *fields, RedisModuleString *s) {
```

Comment:
Should we just use RedisModuleString because nothing is using const char directly?